File: RomantoInteger/RomantoInteger.py
def romanToInt(s: str) -> int:
    sv = {
        "I": 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000
    }

    ans = 0

    for i in range(len(s)):
        if i < len(s)-1 and sv[s[i]] < sv[s[i+1]]:
            ans -= sv[s[i]]
        else:
            ans += sv[s[i]]

    return ans

if __name__ == "__main__":
    roman = "LVIII"     # 58
    print(romanToInt(roman))
    roman = "MCMXCIV"   # 1994
    print(romanToInt(roman))

File: RomantoInteger/RomantoInteger.cpp
#include<iostream>
#include <unordered_map>
using namespace std;

int romanToInt(string s) {
    unordered_map<char, int> sv;
    sv['I'] = 1;
    sv['V'] = 5;
    sv['X'] = 10;
    sv['L'] = 50;
    sv['C'] = 100;
    sv['D'] = 500;
    sv['M'] = 1000;

    int ans = 0;

    for (int i=0; i<s.length(); i++){
        if (sv[s[i]] < sv[s[i+1]]) ans -= sv[s[i]];
        else    ans += sv[s[i]];
    }
    return ans;
}

int main(){
    string s = "MCMXCIV";   // 1994
    cout << romanToInt(s) << endl;
    return 0;
}